servers/navigation_server_3d.cpp

```diff
@@ -121,3 +123,164 @@ NavigationServer3D *NavigationServer3DManager::new_default_server() {
 	ERR_FAIL_COND_V(create_callback == nullptr, nullptr);
 	return create_callback();
 }
+
+///////////////////////////////////////////////////////
+
+Ref<NavigationPathQueryResult3D> NavigationServer3D::query_path(const Ref<NavigationPathQueryParameters3D> &p_query_parameters) const {
+	gd::PathQueryResult _query_result = _query_path(p_query_parameters->get_parameters());
+
+	Ref<NavigationPathQueryResult3D> query_result(memnew(NavigationPathQueryResult3D));
+
+	query_result->set_path(_query_result.path);
+	query_result->set_path_length(_query_result.path_length);
+
+	return query_result;
+}
+
+void NavigationPathQueryParameters3D::set_pathfinding_algorithm(const NavigationPathQueryParameters3D::PathfindingAlgorithm p_pathfinding_algorithm) {
+	switch (p_pathfinding_algorithm) {
+		case PATHFINDING_ALGORITHM_ASTAR: {
+			parameters.pathfinding_algorithm = gd::PathfindingAlgorithm::PATHFINDING_ALGORITHM_ASTAR;
+		} break;
+		default: {
+			parameters.pathfinding_algorithm = gd::PathfindingAlgorithm::PATHFINDING_ALGORITHM_ASTAR;
```

Comment:
I guess we can hit default iff new, unhandled algorithm was added. IMO it's worth putting warning about "not implemented" case here

Reply:
Right, I guess a WARN_PRINT_ONCE here might not hurt in case default is hit.